src/ListPool/ValueListPool.cs

```diff
@@ -204,7 +204,30 @@ public void AddRange(ReadOnlySpan<T> items)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void AddRange(T[] array) => AddRange(array.AsSpan());
+        public void AddRange(T[] array)
+        {
+            int count = Count;
+            T[] disposableBuffer = _disposableBuffer;
+            Span<T> buffer = _buffer;
+
+            bool isCapacityEnough = buffer.Length - array.Length - count > 0;
+            if (!isCapacityEnough)
+            {
+                GrowBuffer(buffer.Length + array.Length);
+                disposableBuffer = _disposableBuffer;
```

Comment:
Can we **short circuit** here? maybe if this condition is true we know that  `_disposableBuffer ` is not null after the resize, so we can do the array copy here and avoid the next if statement

Reply:
done :)